test/compose.test.js

```diff
@@ -16,7 +16,7 @@ describe('compose', function () {
     assert.equal(fn(false, 2), 'B');
     assert.equal(fn('str'), 'C');
     // FIXME: should return correct error message
-    assert.throws(function () {fn()},           /TypeError: Too few arguments in function unnamed \(expected: number or string or boolean, index: 0\)/);
+    assert.throws(function () {fn()},           /TypeError: Too few arguments in function unnamed \(expected: string or number or boolean, index: 0\)/);
     assert.throws(function () {fn(1,2,3)},      /TypeError: Unexpected type of argument in function unnamed \(expected: boolean, actual: number, index: 1\)/);
```

Comment:
This is a similar case; the last signature in the definition got sorted first because it is the only one with just a single (non-rest) argument, changing the order of the possible types for position 0. I don't understand the FIXME comment though; what is wrong with the error message? If you can let me know what the message should have been, happy to improve it.